ScreenToGif/Capture/BaseCapture.cs

```diff
@@ -147,23 +155,51 @@ public virtual Task<int> ManualCaptureAsync(FrameInfo frame, bool showCursor = f
             return showCursor ? CaptureWithCursorAsync(frame) : CaptureAsync(frame);
         }
 
+        /// <summary>
+        /// Safely release all the managed things.
+        /// </summary>
         public virtual async Task Stop()
         {
             if (!WasStarted)
                 return;
 
-            //Stop the consumer thread.
-            BlockingCollection.CompleteAdding();
-
-            await Task.WhenAll(_task);
+            using (_task)
+            {
+                // Stop adding things into the BlockingCollection
+                // and mark it finished
+                BlockingCollection.CompleteAdding();
+                await _task;
+                using (BlockingCollection) { }
```

Comment:
Just call `BlockingCollection.Dispose`?
It should be set to `null` too.

Reply:
OK ;)